Validator/src/main/java/org/openzen/zenscript/validator/visitors/ImplementationCheckValidator.java

```diff
@@ -0,0 +1,121 @@
+package org.openzen.zenscript.validator.visitors;
+
+import org.openzen.zenscript.codemodel.compilation.CompileErrors;
+import org.openzen.zenscript.codemodel.definition.InterfaceDefinition;
+import org.openzen.zenscript.codemodel.identifiers.MethodSymbol;
+import org.openzen.zenscript.codemodel.member.*;
+import org.openzen.zenscript.codemodel.type.TypeID;
+import org.openzen.zenscript.validator.Validator;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ImplementationCheckValidator implements MemberVisitor<Void> {
+
+	private final Validator validator;
+	private final Set<MethodSymbol> implementedMethods = new HashSet<>();
+	private final InterfaceDefinition interfaceDefinition;
+
+	public ImplementationCheckValidator(Validator validator, ImplementationMember implementationMember) {
+		this.validator = validator;
+
+		this.interfaceDefinition = implementationMember.asImplementation()
+				.flatMap(TypeID::asDefinition)
+				.map(d -> d.definition)
+				.filter(InterfaceDefinition.class::isInstance)
+				.map(InterfaceDefinition.class::cast)
+				.orElseThrow(() -> new NoSuchElementException("Must be an implementation!"));
```

Comment:
💭 Not the best way to go about this but at least we get the interfaceDefinition this way I guess?